### DISCUSSION TITLE
LED実装後、キーボードが認識されなくなった

### DISCUSSION THREAD

**Discussion by hanjuku1** [Yowkees/keyball#407] [Q&A]
始めてご質問させていただきます。
keyball61を購入させていただき、昨日作製しておりました。

■ご質問
LED実装後、左手のプレートはLED無事点灯、キーボードも認識しています。
右手プレートがLEDはおろか、キーボードとしても認識しなくなってしましました。
肉眼でショートしている場所は発見できず、どこを重点的にチェックすべきか
ご教授いただければ幸いです。

■経緯
開示いただいている、ビルドガイドに沿って、
[3-8．LED（オプション）のはんだ付け]
まで行ったところです。
3－7で一旦PCに接続し、キーボードとして認識、かついくつかのキーにて動作確認はいたしました。
そのごLEDを実装後、認識もしなくなった。
という状況です。

どこかショートなどしているのではと
あちらこちらフラックスを塗りながら、半田を溶かし直す
ということをしましたが、復活せず。
目視ではよく確認しましたが、ショートしている場所は発見できておりません。
簡易なテスターは保有しているので、電圧や抵抗は見ることが出来ます。

どこを疑って、どう確認するべきかご教授いただければ幸いです。

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

私も経験があるのですが、LEDがショートモードで故障している可能性が高いです。
確認方法ですが、下記写真の赤線部分をカッターの刃の背中側のとがった部分でカリカリと削り、パターンをカットしてください。これにより全LEDのVCCをProMicroから切り離すことができます。これでProMicroが起動できると思います。
LEDを復活させるにはジャンパーの要領で半田で左右をショートさせます。

![61R_LEDcut](https://github.com/Yowkees/keyball/assets/84209772/1cfd9957-be3c-4c65-98fe-ed436d2ee8bf)

次にどのLEDが故障しているのか調べるには、やはり１つずつ除去して探すしかありません。半田こてを２本使用するとチョンチョンと次々に除去できますので楽だと思います。

LEDは高温と湿度に弱いです。小手先を280度以下にして素早く半田付けすることで故障を防げます。４端子を連続して半田付けせず、１端子ずつ時間を置く事も重要です。
また湿度に弱いので、袋から出したらすぐに半田付けすることをお勧めします。

お手数ですが、ご対応お願いいたします。

**Reply by hanjuku1** [reply]
早々のご返信、誠にありがとうございます。LEDCUTを行ったら、キーボードとして認識するようになりました。
そうですか、ＬＥＤの故障ですか。
半田小手が一つしか無いので、どうしたものか、、、チェック方法があればいいのですが。
一つ一つ取り外してみます。
ご連絡ありがとうございました。

**Comment by Yowkees** [comment]
ひとまずProMicroが起動するようになったようで良かったです。
LEDはちょっと根気が必要になりますね。もう一本のはんだごては安価なもので結構ですので、２本あるといろいろと作業の幅が広がってよいと思いますのでご検討ください。
